File: scripts/take_photo_client.py
#!/usr/bin/env python
import rospy
from streaming_media.srv import *

def main():
    rospy.init_node('take_photo_client')

    try:
        rospy.wait_for_service("/take_photo")
        client = rospy.ServiceProxy('/take_photo', TakePhoto)

        res = client(True)
        rospy.loginfo("Photo taken: %s" % res.result)

    except rospy.ServiceException, e:
       rospy.loginfo("Service call failed: %s" % e)


if __name__ == '__main__':
    main()

File: src/take_photo_client.cpp
#include <ros/ros.h>
#include "streaming_media/TakePhoto.h"  // streaming_media 包名

int main(int argc, char** argv) {
    ros::init(argc, argv, "take_photo_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<streaming_media::TakePhoto>("/take_photo");
    streaming_media::TakePhoto srv;

    srv.request.record = true;  // Set to true to take a photo
    if (client.call(srv)) {
        ROS_INFO("Photo taken: %s", srv.response.result.c_str());
    } else {
        ROS_ERROR("Failed to call service /take_photo");
    }
    ros::spin();

    return 0;
}
